pkg/lib/audit/read_store.go

```diff
@@ -34,8 +37,38 @@ func (o QueryPageOptions) Apply(q db.SelectBuilder) db.SelectBuilder {
 		q = q.Where("activity_type = ANY (?)", pq.Array(o.ActivityTypes))
 	}
 
-	if len(o.UserIDs) > 0 {
+	q = o.applyQueryStringFilter(q)
+	return q
+}
+
+func (o QueryPageOptions) applyQueryStringFilter(q db.SelectBuilder) db.SelectBuilder {
+	hasUserIDs := len(o.UserIDs) > 0
+	hasEmail := len(o.EmailAddresses) > 0
+	hasPhone := len(o.PhoneNumbers) > 0
+
+	mergedEmailsAndPhones := o.EmailAddresses
+	mergedEmailsAndPhones = append(mergedEmailsAndPhones, o.PhoneNumbers...)
+	mergedEmailsAndPhones = slice.Deduplicate[string](mergedEmailsAndPhones)
+
+	switch {
+	case hasUserIDs && hasEmail && hasPhone:
+		q = q.Where("(user_id = ANY (?) OR data->'payload'->>'recipient' = ANY (?))", pq.Array(o.UserIDs), pq.Array(mergedEmailsAndPhones))
```

Comment:
Rewrite it to `data #>> '{payload,recipient}'`, and rewrite the index to match this expression.

---

Your explain analysis does not seem to show the new index is actually used.

Reply:
> Your explain analysis does not seem to show the new index is actually use

as per offline discussion, actually used